src/Orchard.Web/Modules/Orchard.OpenId/InMemoryCache.cs

```diff
@@ -1,13 +1,12 @@
 ﻿using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading;
 using Microsoft.IdentityModel.Clients.ActiveDirectory;
 
 namespace Orchard.OpenId {
     public class InMemoryCache : TokenCache {
-        private static Dictionary<string, byte[]> _inMemoryTokenCache;
-        private static ReaderWriterLockSlim _sessionLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+        private static ConcurrentDictionary<string, byte[]> _inMemoryTokenCache;
```

Comment:
Guids generated and assigned to the user by the identity provider. This id will be sent back by the provider after authenticating the user inside the JWT.